ios/Classes/Module/FacebookModule.m

```diff
@@ -157,11 +157,21 @@ - (void)logPurchase:(NSArray<id> *_Nonnull)purchase
 {
   ENSURE_TYPE([purchase objectAtIndex:0], NSNumber);
   ENSURE_TYPE([purchase objectAtIndex:1], NSString);
+  ENSURE_TYPE([purchase objectAtIndex:2], NSDictionary);
 
   NSNumber *amount = [purchase objectAtIndex:0];
   NSString *currency = [TiUtils stringValue:[purchase objectAtIndex:1]];
+  NSDictionary *parameters = [purchase objectAtIndex:2];
 
-  [FBSDKAppEvents logPurchase:[amount doubleValue] currency:currency];
+  // [FBSDKAppEvents logPurchase:[amount doubleValue] currency:currency];
+  [FBSDKAppEvents logPurchase:[amount doubleValue] currency:currency parameters:parameters];
+}
+
+- (void)logRegistrationCompleted:(NSString *_Nonnull)registrationMethod
+{
+    NSDictionary *params = @{FBSDKAppEventParameterNameRegistrationMethod: registrationMethod};
+    
+    [FBSDKAppEvents logEvent:FBSDKAppEventNameCompletedRegistration parameters:params];
```

Comment:
This can be done by reusing the existing log method + the key `fb_registration_method` to keep it generic. Same for Android.